internal/alloydbutil/engine.go

```diff
@@ -27,47 +26,46 @@ func NewPostgresEngine(ctx context.Context, opts ...Option) (*PostgresEngine, er
 	if err != nil {
 		return nil, err
 	}
-	username, usingIAMAuth, err := getUser(ctx, cfg)
+	user, usingIAMAuth, err := getUser(ctx, cfg)
 	if err != nil {
 		return nil, fmt.Errorf("error assigning user. Err: %w", err)
 	}
 	if usingIAMAuth {
-		cfg.user = username
-		token, err := getIAMToken(ctx, username)
-		if err != nil {
-			return nil, err
-		}
-		cfg.password = token
+		cfg.user = user
```

Comment:
If using IAM auth, we should continue to use the IAM user from the environment.
The options are 1) use user and password 2) use user from environment 3) override user in environment with argument "iam_account_email"